rule.raml

```diff
@@ -0,0 +1,364 @@
+#%RAML 0.8
+title: Rule
+version: v0.1.1-20180620
+documentation:
+
+  - title: OCF Copyright
+    content: Copyright (c) 2018 Open Connectivity Foundation, Inc. All rights reserved.
+  - title: OCF License
+    content: !include LICENSE.md
+  - title: OCF Disclaimer
+    content: !include DISCLAIMER.md
+
+
+schemas:
+  - rule-rw: !include schemas/oic.rule-rw-schema.json
+  - baseline: !include schemas/oic.core-schema.json
+  - batch: !include schemas/oic.batch-retrieve-schema.json
+  - linklist: !include schemas/oic.oic-linklist-schema.json
+  - linkupdate: !include schemas/oic.if.linkupdate-schema.json
+  - create: !include schemas/oic.if.create-schema.json
+  - create-response: !include schemas/oic.if.create-response-schema.json
+  - delete-Response: !include schemas/oic.if.delete-response-schema.json
+
+traits:
+  - interface:
+      queryParameters:
+        if:
+          enum: ["oic.if.rw", "oic.if.ll", "oic.if.b", "oic.if.baseline", "oic.if.create", "oic.if.delete", "oic.if.linkupdate"]
+
+
+/RuleURI?if=oic.if.baseline:
+    description: |
+      Baseline representation of a rule contains its common properties
+    displayName: rule-baseline
+
+    get:
+        description: |
+          Common properties of the rule resource type
+        responses:
+          200:
+            body:
+              application/json:
+                schema: baseline
+                example: |
+                {
+                  "rt": ["oic.r.rule"],
+                  "if": ["oic.if.rw", "oic.if.baseline", "oic.if.ll", "oic.if.b", "oic.if.linkupdate", "oic.if.create", "oic.if.delete" ],
+                  "rts": ["oic.r.temperature", "oic.r.switch.binary", "oic.r.scenevalue" ],
+                  "n": "Thermostat heat-on rule"
+                }
+
+
+/RuleURI?if=oic.if.ll:
+    description: |
+      Link list representation of a rule contains its links and bindings
+    displayName: rule-linklist
+
+    get:
+        description: |
+          Links of the rule resource type
+        responses:
+          200:
+            body:
+              application/json:
+                schema: linklist
+                  example: |
+                    [
+                      {
+                        "rel": ["self", "item"]
```

Comment:
Why is this self Link here?  It doesn't seem to add any value.  So the Rule is a link within the Rule?  If it is here, why are there no other Link Parameters included as well?